tests/instancetype/upgrade.go

```diff
@@ -0,0 +1,599 @@
+
+			By("Waiting for the upgrade to be marked as successful")
+			Eventually(func(g Gomega) {
+				crUpgrade, err := virtClient.ControllerRevisionUpgrade(vm.Namespace).Get(context.Background(), crUpgrade.Name, metav1.GetOptions{})
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(crUpgrade.Status).ToNot(BeNil())
+				g.Expect(crUpgrade.Status.Phase).ToNot(BeNil())
+				g.Expect(*crUpgrade.Status.Phase).To(Equal(instancetypev1beta1.UpgradeSucceeded))
+			}, 30*time.Second, time.Second).Should(Succeed())
+
+			By("asserting that the ControllerRevision referenced by the VirtualMachine has been updated to the latest version")
+			vmRevisionName := getVMRevisionName()
+
+			crUpgrade, err = virtClient.ControllerRevisionUpgrade(vm.Namespace).Get(context.Background(), crUpgrade.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(crUpgrade.Status.Result.Name).To(Equal(vmRevisionName))
+			Expect(crUpgrade.Status.Result.Version).To(Equal(instancetypeapi.LatestVersion))
+
+			cr, err = virtClient.AppsV1().ControllerRevisions(vm.Namespace).Get(context.Background(), vmRevisionName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			// If a new CR has been created assert that the old CR has been deleted
+			if originalCRName != vmRevisionName {
+				_, err := virtClient.AppsV1().ControllerRevisions(vm.Namespace).Get(context.Background(), originalCRName, metav1.GetOptions{})
+				Expect(err).Should(HaveOccurred())
+				Expect(errors.ReasonForError(err)).Should(Equal(metav1.StatusReasonNotFound))
+			}
+
+			Expect(cr.Labels).To(HaveKeyWithValue(instancetypeapi.ControllerRevisionObjectVersionLabel, instancetypeapi.LatestVersion))
+
+			decodedObj, err := runtime.Decode(generatedscheme.Codecs.UniversalDeserializer(), cr.Data.Raw)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(decodedObj.GetObjectKind().GroupVersionKind().Version).To(Equal(instancetypeapi.LatestVersion))
+		},
+			Entry("VirtualMachineInstancetype from v1beta1 without labels to latest",
+				func() (*appsv1.ControllerRevision, error) {
+					instancetype := &instancetypev1beta1.VirtualMachineInstancetype{
+						ObjectMeta: metav1.ObjectMeta{
+							GenerateName: "instancetype",
+						},
+						Spec: instancetypev1beta1.VirtualMachineInstancetypeSpec{
+							CPU: instancetypev1beta1.CPUInstancetype{
+								Guest: uint32(1),
+							},
+							Memory: instancetypev1beta1.MemoryInstancetype{
+								Guest: resource.MustParse("128Mi"),
+							},
+						},
+					}
+					instancetype, err := virtClient.VirtualMachineInstancetype(util.NamespaceTestDefault).Create(context.Background(), instancetype, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					cr, err := createControllerRevision(instancetype)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(cr.Labels).To(HaveKey(instancetypeapi.ControllerRevisionObjectVersionLabel))
+					delete(cr.Labels, instancetypeapi.ControllerRevisionObjectVersionLabel)
```

Comment:
This will not update the CR on the API which the UpgradeController is going to look at?